src/com/ableton/Pyenv.groovy

```diff
@@ -8,6 +8,28 @@ package com.ableton
  * @see com.ableton.VirtualEnv
  */
 class Pyenv implements Serializable {
+  /**
+   * Script context.
+   * <strong>Required value, may not be null!</strong>
+   */
+  Object script
+  /**
+   * Python version, as listed in {@code pyenv install --list}.
+   * <strong>Required value, may not be null!</strong>
+   */
+  String pythonVersion
+  /**
+   * Pyenv root directory.
+   * <strong>Required value, may not be null!</strong>
+   */
+  String pyenvRoot
+
+  Pyenv(Object script, String pythonVersion, String pyenvRoot) {
+    this.script = script
```

Comment:
Shouldn't we assert that the values passed in aren't null here?

Reply:
No, we don't assert in constructors. I don't think we've formally documented this practice, but I don't think it's a good practice since most users don't expect creation of a new object to fail and seldom surround it with `try/catch`. We assert these values at usage.